util/rlp/src/untrusted_rlp.rs

```diff
@@ -371,7 +371,8 @@ impl<'a> BasicDecoder<'a> {
 				}
 				let len = decode_usize(&bytes[1..begin_of_value])?;
 
-				let last_index_of_value = begin_of_value + len;
+				let last_index_of_value = begin_of_value.overflowing_add(len)
```

Comment:
does this compile? i thought `overflowing_add` returns `(usize, bool)`. `checked_add` returns an `Option`, though.